### PR TITLE
CORDA-1932 Fixing network map certificate path verification

### PR DESCRIPTION
**Pull request by mkit**
This PR fixes the certificate path verification process in case of the incoming network map data. It no longer assumes any specific certificate path length.

### CODE REVIEW

File: core/src/main/kotlin/net/corda/core/internal/DigitalSignatureWithCert.kt [hunk 1]
```diff
@@ -7,16 +7,20 @@ import net.corda.core.serialization.CordaSerializable
...
 // TODO: Rename this to DigitalSignature.WithCert once we're happy for it to be public API. The methods will need documentation
 // and the correct exceptions will be need to be annotated
 /** A digital signature with attached certificate of the public key. */
-class DigitalSignatureWithCert(val by: X509Certificate, bytes: ByteArray) : DigitalSignature(bytes) {
+open class DigitalSignatureWithCert(val by: X509Certificate, bytes: ByteArray) : DigitalSignature(bytes) {
     fun verify(content: ByteArray): Boolean = by.publicKey.verify(content, this)
     fun verify(content: OpaqueBytes): Boolean = verify(content.bytes)
 }
 
+/** A digital signature with attached certificate path. The first certificate in the path corresponds to the data signer key. */
```
Comment (fenryka) [inline core/src/main/kotlin/net/corda/core/internal/DigitalSignatureWithCert.kt]: If we're doing Java Docs lets list the properties as well

File: node-api/src/main/kotlin/net/corda/nodeapi/internal/network/NetworkMap.kt [hunk 2]
```diff
@@ -58,19 +56,12 @@ data class ParametersUpdate(
 )
 
 /** Verify that a Network Map certificate is issued by Root CA and its [CertRole] is correct. */
-// TODO: Current implementation works under the assumption that there are no intermediate CAs between Root and
-//      Network Map. Consider a more flexible implementation without the above assumption.
-
 fun <T : Any> SignedDataWithCert<T>.verifiedNetworkMapCert(rootCert: X509Certificate): T {
     require(CertRole.extract(sig.by) == CertRole.NETWORK_MAP) { "Incorrect cert role: ${CertRole.extract(sig.by)}" }
-    X509Utilities.validateCertificateChain(rootCert, sig.by, rootCert)
+    if (this.sig is DigitalSignatureWithCertPath) {
```
Comment (fenryka) [inline node-api/src/main/kotlin/net/corda/nodeapi/internal/network/NetworkMap.kt]: do we need this?

**Comment by fenryka** [comment]
This is going into Corda so can we move the ticket into the Corda Namespace in Jira

File: core/src/main/kotlin/net/corda/core/internal/DigitalSignatureWithCert.kt (hunk 1 above)
Comment (mkit) [reply core/src/main/kotlin/net/corda/core/internal/DigitalSignatureWithCert.kt]: Done.

**Pull request merged**